ncm-network/src/main/perl/network.pm

```diff
@@ -1481,10 +1489,30 @@ sub stop
     if ($nwupdated || @ifaces) {
         if (@ifaces) {
             my @cmds;
+            # delete NetworkManager connections
+            foreach my $file ( glob "/etc/NetworkManager/system-connections/*.nmconnection" ) {
+                push(@cmds, ["rm", "-f", $file]);
+            }
             foreach my $iface (@ifaces) {
-                # how do we actually know that the device was up?
-                # eg for non-existing device eth4: /sbin/ifdown eth4 --> usage: ifdown <device name>
-                push(@cmds, ["/sbin/ifdown", $iface]);
+                if ($self->_is_executable($IFDOWN_CMD)) {
+                    # how do we actually know that the device was up?
+                    # eg for non-existing device eth4: /sbin/ifdown eth4 --> usage: ifdown <device name>
+                    push(@cmds, [$IFDOWN_CMD, $iface]);
+                } elsif (defined($allow_nm) && $allow_nm) {
+                    # get NetworkManager connection name
+                    my $nm_con = $self->runrun(["nmcli", "-g", "GENERAL.CONNECTION", "device", "show", $iface]);
+                    chomp($nm_con);
+                    if ($nm_con ne "" ) {
+                        # delete the NetworkManager connection
+                        push(@cmds, ["nmcli", "connection", "delete", $nm_con]);
+                    } else {
+                        # stop the interface
+                        push(@cmds, ["ip", "link", "set", $iface, "down"]);
+                    }
+                } else {
+                    # stop the interface
```

Comment:
at the very least log an error here. because this is not permanent removal

Reply:
can you please elaborate? this is just trying to reset the interface to a default state so the new configuration can be applied with a greater chance of success